File: src/main/java/com/example/wizytydomowe/Address/Address.java
package com.example.wizytydomowe.Address;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.Pattern;

@Entity
@Data
@Table(name = "address")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String city;
    @Pattern(regexp = "^[0-9]{2}-[0-9]{3}")
    private String zipCode;
    private String street;
    private Integer houseNumber;
    private Integer flatNumber;
    private Float longitude;
    private Float latitude;
}

File: uiw/ui/src/app/address.ts
export class Address {
  city: string = '';
  zipCode: string = '';
  street: string = '';
  houseNumber: number = 23;
  flatNumber: number = 13;
  longitude: number = 15.887;
  latitude: number = 50.054;

  constructor(city: string, zipCode: string, street: string, houseNumber: number, flatNumber: number, longitude: number,
              latitude: number) {
    this.city = city;
    this.zipCode = zipCode;
    this.street = street;
    this.houseNumber = houseNumber;
    this.flatNumber = flatNumber;
    this.longitude = longitude;
    this.latitude = latitude;
  }
}

